main.py

```diff
@@ -1,112 +1,182 @@
-# The cookie will be stored 30 days long.
+# Customize the cookie name, value and cookie expire date.
+# The cookie will be stored on the client site for 30 days long.
 COOKIE_NAME = "OldTamil"
+COOKIE_EXPIRE = datetime.datetime.now() + datetime.timedelta(days=30)
 COOKIE_VALUE = 0
-COOKIE_LIFE_TIME = 24 * 60 * 60 * 30   
-
 
-# User credentials.
-USERNAME = 'Hippod'
-PASSWORD = 'admin'
 
 # Define the required host and port.
 HOST = '127.0.0.1'
-PORT = 8888
+PORT = 8080
+
+# All html and other files required from outside.
+# All these files required by the server, modifying contents inside the file
+# may cause server error.
+LOGIN_HTML = 'login.html'
+SITE_HTML = 'site.html'
+REDIRECT_HTML = 'redirect.html'
+CONFIG_FILE = 'configuration.json'
+
+
+# Simple error web view for server error.
+# Server error can be caused by missing required files.
+ERROR = '''<!DOCTYPE html>
+<html>
+  <head>
+    <title>SERVER ERROR</title>
+    <style type ="text/css">
+    body { text-align:center; padding: 10%;
+    font-weight: bold; font: 20px Helvetica, sans-serif; }
+    a {padding: 30px;}
+    h1 {color:#D0AA0E ; font-family: Arial, Helvetica, sans-serif;
+        font-size: 30px;}
+    </style>
+  </head>
+     <body>
+       <h1> Unexpected server error </h1>
+       <p> Please try again later</p>
+     </body>
+</html>'''
+
+
+# Check the requested file by user, and return server error if the file is
+# missing.
+def check_the_file(filename):
+    if not os.path.isfile(filename):
+        print("Internal server error! file not found: {}".format(filename))
+        return False
+    response = open(filename).read()
+    return response
 
 
+# Checking the configuration file and valid the data.
+# The checking may fail by missing json file, missing valid data in the file.
+# Cause server error when configuration file has been modified.
+def json_reader(filename):
+    check_file = check_the_file(filename)
+    if not check_file:
+        return False
+    configure_file = open(filename)
+    load_data = json.load(configure_file)
+    try:
+        if not(load_data['USERNAME'] and load_data['PASSWORD']):
+            return False
+        return load_data
+    except KeyError:
+        print("{} doesn't have valid keys and value".format(filename))
+        pass
+
+
+# Identifying the user by validate the cookie. User must login if there is no
+# valid cookie
+# Cookie safety has been implemented. If user tries to
+# modify the cookie value it returns False.
+# Make sure user cannot modify cookie expire date.
 def check_cookie(request):
-    """cookie_check!
-    This function will check for valid cookie on all pages!
-    Return True if there is valid cookie.
-
-    """
     cookie = request.cookies.get(COOKIE_NAME, None)
     if not cookie:
         return False
-    if COOKIE_VALUE == 0:
+    try:
+        if not int(cookie) == COOKIE_VALUE:
+            return False
         return True
+    except ValueError:
```

Comment:
we might not need this here. anyway, its not working. what is the intention here? do you want an error log for the developer? think we can remove this...